netty-mqtt/src/main/java/org/thingsboard/mqtt/MqttClientImpl.java

```diff
@@ -426,7 +426,12 @@ public void disconnect() {
         disconnected = true;
         if (this.channel != null) {
             MqttMessage message = new MqttMessage(new MqttFixedHeader(MqttMessageType.DISCONNECT, false, MqttQoS.AT_MOST_ONCE, false, 0));
-            this.sendAndFlushPacket(message).addListener(future1 -> channel.close());
+            ChannelFuture channelFuture = this.sendAndFlushPacket(message);
+            eventLoop.schedule(() -> {
+                if (!channelFuture.isDone()) {
+                    this.channel.close();
+                }
+            }, 50, TimeUnit.MILLISECONDS);
```

Comment:
Could you please describe why we need to do this in this way? Why 50 ms, and so on?